source/common/filter/config_discovery_impl.cc

```diff
@@ -132,19 +132,17 @@ void FilterConfigSubscription::onConfigUpdate(
   for (auto* provider : filter_config_providers_) {
     provider->validateTerminalFilter(filter_config_name_, factory.name(), is_terminal_filter);
   }
-  Envoy::Http::FilterFactoryCb factory_callback =
-      factory.createFilterFactoryFromProto(*message, stat_prefix_, factory_context_);
```

Comment:
This can throw. Is there any implication on moving this into config providers? Does the exception propagate to xDS rejection correctly?